logger/tests/integration_tests.rs

```diff
@@ -1,465 +1,180 @@
-use std::net::{Ipv4Addr, SocketAddr};
+use std::{
+    net::{Ipv4Addr, SocketAddr},
+    time::{Duration, SystemTime},
+};
 
-use opentelemetry::KeyValue;
-use opentelemetry_otlp::WithExportConfig;
-use opentelemetry_proto::tonic::collector::trace::v1::trace_service_server::TraceServiceServer;
 use portpicker::pick_unused_port;
 use pretty_assertions::assert_eq;
-use serde_json::{json, Value};
-use shuttle_common::{
-    claims::Scope,
-    tracing::{FILEPATH_KEY, LINENO_KEY, MESSAGE_KEY, NAMESPACE_KEY, TARGET_KEY},
-};
+use prost_types::Timestamp;
+use shuttle_common::claims::Scope;
 use shuttle_common_tests::JwtScopesLayer;
-use shuttle_logger::{Service, ShuttleLogsOtlp, Sqlite};
+use shuttle_logger::{Service, Sqlite};
 use shuttle_proto::logger::{
-    logger_client::LoggerClient, logger_server::LoggerServer, LogItem, LogLevel, LogsRequest,
+    logger_client::LoggerClient, logger_server::LoggerServer, FetchedLogItem, LogsRequest,
+    StoreLogsRequest, StoredLogItem,
 };
 use tokio::time::timeout;
 use tonic::{transport::Server, Request};
-use tracing::{debug, error, info, instrument, trace, warn};
-use tracing_subscriber::prelude::*;
 
-// TODO: find out why these tests affect one-another. If running them together setting the timeouts
-// low will cause them to fail spuriously. If running single tests they always pass.
+const SHUTTLE_SERVICE: &str = "test";
+
 #[tokio::test]
-async fn generate_and_get_runtime_logs() {
+async fn store_and_get_logs() {
     let port = pick_unused_port().unwrap();
     let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port);
     const DEPLOYMENT_ID: &str = "runtime-fetch-logs-deployment-id";
 
     // Start the logger server in the background.
-    tokio::task::spawn(async move {
-        let sqlite = Sqlite::new_in_memory().await;
-
+    let sqlite = Sqlite::new_in_memory().await;
+    let sqlite_clone = sqlite.clone();
+    let server = tokio::task::spawn(async move {
         Server::builder()
             .layer(JwtScopesLayer::new(vec![Scope::Logs]))
-            .add_service(TraceServiceServer::new(ShuttleLogsOtlp::new(
-                sqlite.get_sender(),
+            .add_service(LoggerServer::new(Service::new(
+                sqlite_clone.get_sender(),
+                sqlite_clone,
             )))
-            .add_service(LoggerServer::new(Service::new(sqlite.get_sender(), sqlite)))
             .serve(addr)
             .await
             .unwrap()
     });
```

Comment:
We can extract this to a function called `spawn_server` and avoid having code repetitions for tests.

Reply:
Done.